backend/service/chaos/experimentation/experimentstore/expriment_run_details.go

```diff
@@ -53,53 +59,53 @@ func NewRunDetails(fetchedID uint64, creationTime time.Time, startTime sql.NullT
 	return runConfigPair, err
 }
 
-func timeToString(t sql.NullTime) string {
-	layout := "01/02/06 15:04:05"
-	if t.Valid {
-		return t.Time.Format(layout)
-	} else {
-		return "Undefined"
-	}
-}
-
-func timesToStatus(startTime sql.NullTime, endTime sql.NullTime, cancellationTime sql.NullTime, now time.Time) experimentation.Status {
+func timesToStatus(startTime sql.NullTime, endTime sql.NullTime, cancellationTime sql.NullTime, now time.Time) (experimentation.ExperimentRunDetails_Status, error) {
 	if !startTime.Valid {
-		return experimentation.Status_UNSPECIFIED
+		return experimentation.ExperimentRunDetails_UNSPECIFIED, errors.New("experiment doesn't have startTime")
 	}
 
 	if cancellationTime.Valid {
 		if cancellationTime.Time.After(startTime.Time) {
 			if endTime.Valid {
-				return experimentation.Status_STOPPED
+				return experimentation.ExperimentRunDetails_STOPPED, nil
 			} else {
-				return experimentation.Status_COMPLETED
+				return experimentation.ExperimentRunDetails_COMPLETED, nil
 			}
 		} else {
-			return experimentation.Status_CANCELED
+			return experimentation.ExperimentRunDetails_CANCELED, nil
 		}
 	} else {
 		if now.Before(startTime.Time) {
-			return experimentation.Status_SCHEDULED
+			return experimentation.ExperimentRunDetails_SCHEDULED, nil
 		} else if now.After(startTime.Time) && (!endTime.Valid || now.Before(endTime.Time)) {
-			return experimentation.Status_RUNNING
+			return experimentation.ExperimentRunDetails_RUNNING, nil
 		}
-		return experimentation.Status_COMPLETED
+		return experimentation.ExperimentRunDetails_COMPLETED, nil
+	}
+}
+
+func timeToString(t sql.NullTime) string {
+	layout := "01/02/06 15:04:05"
+	if t.Valid {
+		return t.Time.Format(layout)
+	} else {
+		return "Undefined"
```

Comment:
Have we checked if Lyft has a go library for DateTime like python? If so, maybe the library has some functionality

Reply:
What functionality? I'm just providing a date format I want to use here (that's the only custom logic here?) but I think that this is fine since I want to be able to specify the format that users see in the UI

Reply:
Discussed offline - we can improve this by making it possible for frontend to send `Date` directly to frontend instead of being forced to convert to string before sending a given value to frontend